src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright 2019 New Relic Corporation. All rights reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package kamon.newrelic.metrics
+
+import com.newrelic.telemetry.SimpleMetricBatchSender
+import com.newrelic.telemetry.metrics.{MetricBatch, MetricBatchSender}
+import com.typesafe.config.Config
+import kamon.Kamon
+import kamon.metric.PeriodSnapshot
+import kamon.module.{MetricReporter, Module, ModuleFactory}
+import kamon.newrelic.AttributeBuddy.buildCommonAttributes
+import org.slf4j.LoggerFactory
+
+import scala.jdk.CollectionConverters._
+
+class NewRelicMetricsReporter(senderBuilder: () => MetricBatchSender = () => NewRelicMetricsReporter.buildSender()) extends MetricReporter {
+
+  private val logger = LoggerFactory.getLogger(classOf[NewRelicMetricsReporter])
+  @volatile private var commonAttributes = buildCommonAttributes(Kamon.config())
+  @volatile private var sender: MetricBatchSender = senderBuilder()
+
+  override def reportPeriodSnapshot(snapshot: PeriodSnapshot) = {
+    logger.warn("NewRelicMetricReporter reportPeriodSnapshot...")
+    val periodStartTime = snapshot.from.toEpochMilli
+    val periodEndTime = snapshot.to.toEpochMilli
+
+    val counters = snapshot.counters.flatMap { counter =>
+      NewRelicCounters(periodStartTime, periodEndTime, counter)
+    }
+    val gauges = snapshot.gauges.flatMap { gauge =>
+      NewRelicGauges(periodEndTime, gauge)
+    }
+    val histogramMetrics = snapshot.histograms.flatMap { histogram =>
+      NewRelicDistributionMetrics(periodStartTime, periodEndTime, histogram, "histogram")
+    }
+    val timerMetrics = snapshot.timers.flatMap { timer =>
+      NewRelicDistributionMetrics(periodStartTime, periodEndTime, timer, "timer")
+    }
```

Comment:
This is missing a conversion for the range samplers, was that on purpose?

Reply:
I think we missed that for our first pass. We can add them pretty easily as a followup PR.